### PR TITLE
Replaces console.log with pino for logging.

### LINKED ISSUE #19: console.log is slow for logging
We should replace it with https://www.npmjs.com/package/pino

### PR DESCRIPTION
**Pull request by mcollina**
Closes #19.

### CODE REVIEW

**Comment by davidmarkclements** [comment]
Obviously I'm 100% for this, however we need to test that pino works in browser still *and* do we really want to load all that into the browser (maybe we should make a pino browser logger thats a super minimal wrapper around pino api)

**Comment by davidmarkclements** [comment]
Cf #17

**Comment by mcollina** [comment]
Maybe we shouldn't log in browsers. Certainly, I would not want to log to console.log
All logging maybe should probably be disabled as a default, and completely opt-in.

**Comment by davidmarkclements** [comment]
I think opt in works, and covers the browser issue - we just have to make sure that not opting in doesn't include logger in browser bundle (e.g require('pi'+'no))

**Comment by davidmarkclements** [comment]
not keen on the `logger` var post-initialisation hoisting

is there a better way?

File: core/transport.js [hunk 1]
```diff
@@ -92,12 +94,16 @@ module.exports = function (driver) {
...
     mu = muInstance
+    logger = mu.log.child({ muid })
   }
 
 
 
   function setId (id) {
     muid = id
+    if (mu) {
+      logger = mu.log.child({ muid })
```
Comment (davidmarkclements) [inline core/transport.js]: must we do this (possibly async) indirection stuff?

File: core/transport.js [hunk 2]
```diff
@@ -92,12 +94,16 @@ module.exports = function (driver) {
 
   function setMu (muInstance) {
     mu = muInstance
+    logger = mu.log.child({ muid })
```
Comment (davidmarkclements) [inline core/transport.js]: this isn't nice - something about the way we're initalizing mu is sending the wrong developer pattern signals

Comment (mcollina) [inline core/transport.js]: the problem is how mu is passed around. Not the logger itself, it's another PR to address that issue.

File: core/transport.js (hunk 1 above)
Comment (mcollina) [inline core/transport.js]: Yes. Until we change how mu is passed around.

**Pull request merged**